clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -309,11 +309,14 @@ private boolean renameDirectory(Path src, Path dst) throws IOException {
             dstFileStatus = getFileStatus(dst);
             dstExists = true;
             if (!dstFileStatus.isDirectory()) {
-                //TODO (Tals): move all of this to the input validation part of the rename functionality with https://github.com/treeverse/lakeFS/issues/2061
-                // Mimics the behaviour in https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1527
-                LOG.error("renameDirectory: src {} is a directory and dst {} is a file", src, dst);
-                return false;
+                // Same as https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1527
+                throw new FileAlreadyExistsException("Failed rename " + src + " to " + dst
+                        + ". destination file already exists.");
             }
```

Comment:
Failing that copy, The S3A Way seems to be to use a semicolon (and along the way avoid the strange noncapitalization of "destination").

```suggestion
                // Same as https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1527
                throw new FileAlreadyExistsException("Failed rename " + src + " to " + dst
                        + "; destination file already exists.");
            }
```

Also below.